Fix select component

Hyperkid123 opened the pull request:
@chalettu Stumbled upon issue with multiple versions of react whiles using select component. 

This is caused by react-select v1. It is using old version of react and old API for refs. That blows up ui in Miq.

Hyperkid123 commented:
@miq-bot assign @karelhala 
@miq-bot add_label bug

karelhala commented on src/forms/finalFormComponent.jsx:
```diff
@@ -5,25 +5,34 @@ import PropTypes from 'prop-types';
+    closeMenuOnSelect={!rest.multi}
+    noOptionsMessage={() => __('No option found')}
+    onChange={option =>
+    input.onChange(rest.multi ? selectValue(option) : option ? option.value : undefined)} // eslint-disable-line no-nested-ternary
```
Please align this line and use shortcut (if applicable)
```
onChange={
  option => input.onChange(rest.multi ? selectValue(option) : option && option.value)
}
```

Hyperkid123 commented:
@karelhala do you have any idea why the tests are busted again? It works locally...

Hyperkid123 commented:
@karelhala oh its the tagging component.
@PanSpagetka do you mind if i remove the snapshot test? Travis can't handle that.

karelhala commented:
I see a lot of outdated snapshots in the build.

Hyperkid123 commented:
i get why most of them changed. The select component was used in them, but the tagging one has over 80k lines.

PanSpagetka commented:
@Hyperkid123 It is ok for me if you remove snapshots

Hyperkid123 commented:
@karelhala ok i've removed a lot of snaphost, something had to change in some dependency, because the snapshots were incredibly bloated. I've replaced them with shallow snapshots...

chalettu commented:
@karelhala , will merging this PR go ahead and bump the react-ui-components version number for the package in NPM?   I am dependent on this fix for a PR we have been working on

karelhala commented:
@chalettu yes, new version of `react-ui-components` is automatically released with new version `3.23.1`

Hyperkid123 commented:
@chalettu it should release new version in a second

karelhala commented:
:tada: This PR is included in version 0.11.7 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@manageiq/react-ui-components)
- [GitHub release](https://github.com/ManageIQ/react-ui-components/releases/tag/v0.11.7)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
